pandas/core/indexes/base.py

```diff
@@ -1376,6 +1393,12 @@ def set_names(self, names, level=None, inplace: bool = False):
         if not is_list_like(names) and level is None and self.nlevels > 1:
             raise TypeError("Must pass list-like as `names`.")
 
+        if is_dict_like(names) and not isinstance(self, ABCMultiIndex):
```

Comment:
all of these conditions can be if/elif

Reply:
Wanted to be consistent with the existing conditions, changed them to elif